ledger/byzcoin/mod.go

```diff
@@ -257,25 +329,103 @@ func (ldgr *Ledger) Watch(ctx context.Context) <-chan ledger.TransactionResult {
 	return results
 }
 
-type actor struct {
-	gossiper gossip.Gossiper
+type actorLedger struct {
+	*Ledger
+	bcActor blockchain.Actor
 }
 
-func newActor(g gossip.Gossiper) actor {
-	return actor{
-		gossiper: g,
+func newActor(l *Ledger, a blockchain.Actor) actorLedger {
+	return actorLedger{
+		Ledger:  l,
+		bcActor: a,
 	}
 }
 
+func (a actorLedger) HasStarted() <-chan error {
+	return a.initiated
+}
+
+func (a actorLedger) Setup(players mino.Players) error {
+	authority, ok := players.(crypto.CollectiveAuthority)
+	if !ok {
+		return xerrors.Errorf("players must implement '%T'", authority)
+	}
+
+	rosterpb, err := a.encoder.Pack(a.governance.rosterFactory.New(authority))
+	if err != nil {
+		return xerrors.Errorf("couldn't pack roster: %v", err)
+	}
+
+	payload := &GenesisPayload{Roster: rosterpb.(*Roster)}
+
+	page, err := a.proc.setup(payload)
+	if err != nil {
+		return xerrors.Errorf("couldn't store genesis payload: %v", err)
+	}
+
+	payload.Footprint = page.GetFootprint()
```

Comment:
I find this section strange: You need a payload to create a page to create a footprint for that same payload. I don't have the whole picture in mind so that may make perfect sense, just noticing it.

Reply:
Basically you apply the transactions of the payload to the inventory to get the footprint back.